src/test/java/com/amazon/opendistroforelasticsearch/sql/unittest/JSONRequestTest.java

```diff
@@ -125,6 +125,139 @@ public void aggWithWhereOnNestedOrNested() {
+        assertThat(
+                query(explainSQL, "/aggregations/name/aggregations/bucket_filter/bucket_selector/buckets_path"),
+                equalTo("{\"count_0\":\"projects@NESTED.count_0\"}"));
+    }
+
+    @Test
+    public void aggInHavingWithWhereOnNested() {
+        JSONObject explainSQL = explainSQLToJson("SELECT name " +
+                                                 "FROM employee " +
+                                                 "WHERE nested(projects.name, 'projects') LIKE '%security%' " +
+                                                 "GROUP BY name " +
+                                                 "HAVING COUNT(nested(projects, 'projects')) > 1");
+        assertThat(
+                query(
+                        explainSQL,
+                        "/aggregations/name/aggregations/projects@NESTED/aggregations/projects@FILTER/aggregations/count_0/value_count"),
+                equalTo("{\"field\":\"_index\"}"));
+        assertThat(
+                query(explainSQL, "/aggregations/name/aggregations/bucket_filter/bucket_selector/buckets_path"),
+                equalTo("{\"count_0\":\"projects@NESTED>projects@FILTER.count_0\"}"));
+        assertThat(
+                query(
+                        explainSQL,
+                        "/aggregations/name/aggregations/projects@NESTED/aggregations/projects@FILTER/filter/bool/must"),
+                equalTo("[{\"wildcard\":{\"projects.name\":{\"boost\":1,\"wildcard\":\"*security*\"}}}]"));
+    }
+
+    @Test
+    public void aggInHavingWithWhereOnParentOrNested() {
+        JSONObject explainSQL = explainSQLToJson("SELECT name " +
+                                                 "FROM employee " +
+                                                 "WHERE name LIKE '%smith%' OR nested(projects.name, 'projects') LIKE '%security%' " +
+                                                 "GROUP BY name " +
+                                                 "HAVING COUNT(nested(projects, 'projects')) > 1");
+        assertThat(
+                query(explainSQL, "/aggregations/name/aggregations/projects@NESTED/aggregations/count_0/value_count"),
+                equalTo("{\"field\":\"_index\"}"));
+        assertThat(
+                query(explainSQL, "/aggregations/name/aggregations/bucket_filter/bucket_selector/buckets_path"),
+                equalTo("{\"count_0\":\"projects@NESTED.count_0\"}"));
+    }
+
+    @Test
+    public void aggInHavingWithWhereOnParentAndNested() {
+        JSONObject explainSQL = explainSQLToJson("SELECT name " +
+                                                 "FROM employee " +
+                                                 "WHERE name LIKE '%smith%' AND nested(projects.name, 'projects') LIKE '%security%' " +
+                                                 "GROUP BY name " +
+                                                 "HAVING COUNT(nested(projects, 'projects')) > 1");
+        assertThat(
```

Comment:
This formatting looks inconsistent

Reply:
Any formatting suggestion? I format the query by making the SQL clause aligend.

Reply:
Sorry, should've been more clear- I was referring to differences such as the ones between lines 202-203 and line 207. On line 202, it appears as if query() lists its arguments on separate lines, whereas on line 207, they are all on the same line.

Reply:
Done.